.github/workflows/generate-go-docs.yaml

```
@@ -9,70 +9,111 @@ on:
         run: |
-          generate-go-function-docs diff -b ${{ steps.pr-details.outputs.base_commit }} -c ${{ steps.pr-details.outputs.head_commit }} -g fake_slash
+          # Add go binary to PATH
+          PATH=$PATH:$(go env GOPATH)/bin
+          export PATH
+          # TODO check why this flag is needed, if it has a default!
+          # TODO we need a way to determine which modules have changes and run this command for all of them
+          generate-go-function-docs diff -b ${{ github.event.pull_request.base.sha }} -c ${{ github.event.pull_request.head.sha }} -g fake_slash -f ./seth --goModLocation ./seth/go.mod
 
       - name: Make sure code still compiles
         run: |
+          # TODO we need a way to determine which modules have changes and run this command for all of them
+          cd seth
           go mod tidy
-          go mod vendor
           go build ./...
 
-      - name: Commit changes
+      - name: Setup branch names
+        id: branch-names
+        shell: bash
         run: |
-          git add .
-          git commit -m "Add automatically generated go documentation"
-          git push origin ${{ steps.pr-details.outputs.pr_branch }}-docs
+          echo "base_branch=${{ github.head_ref }}" >> $GITHUB_OUTPUT
+          echo "new_branch=${{ github.head_ref }}-docs" >> $GITHUB_OUTPUT
+
+      - name: Setup GitHub Token for creating a new PR
+        id: setup-github-token-write
+        uses: smartcontractkit/.github/actions/setup-github-token@9e7cc0779934cae4a9028b8588c9adb64d8ce68c # setup-github-token@0.1.2
+        with:
+          aws-role-arn: ${{ secrets.AWS_ROLE_ARN_CREATE_PR }}
+          aws-lambda-url: ${{ secrets.GATI_LAMBDA_TT_URL }}
+          aws-region: ${{ secrets.AWS_REGION }}          
+
+      # - name: Checkout new branch
+      #   uses: peterjgrainger/action-create-branch@10c7d268152480ae859347db45dc69086cef1d9c # v3.0.0
+      #   env:
+      #     GITHUB_TOKEN: ${{ steps.setup-github-token-write.outputs.access-token }}
+      #   with:
+      #     branch: ${{ steps.branch-names.outputs.new_branch }}
+
+      # - name: Commit changes            
+      #   uses: planetscale/ghcommit-action@13a844326508cdefc72235201bb0446d6d10a85f # v0.1.6
+      #   with:
+      #     commit_message: "[Bot] Add automatically generated go documentation"
+      #     repo: ${{ github.repository }}
+      #     branch: ${{ steps.branch-names.outputs.new_branch }}
+      #     # TODO we need a way to determine which modules have changes and run this command for all of them
+      #     file_pattern: "seth/*"
+      #   env:
+      #     GITHUB_TOKEN: ${{ steps.setup-github-token-write.outputs.access-token }}
+
+      # - name: Fetch repository changes
+      #   uses: actions/checkout@v3
+      #   with:
+      #     fetch-depth: 0            
 
       - name: Create a new PR targeting current PR
-        uses: peter-evans/create-pull-request@v5
+        uses: peter-evans/create-pull-request@v7
```

Since it's a 3rd party action, make sure to pin this to the git sha and comment that sha version like:

```suggestion
        uses: peter-evans/create-pull-request@5e914681df9dc83aa4e4905692ca88beb2f9e91f # v7.0.5
```

---

yes sir! thank you for the reminder :-)